elastic/kibana
# Fix Alerts page history navigation

Fixes #180441

This PR fixes the URL navigation by first calculating the initial state and initializing both the URL and state container with the default value, then starting the sync process. (Special thanks to @Dosant )

https://github.com/elastic/kibana/assets/12370520/87d1a542-d070-4d32-bcd4-aa1565e96acb

This PR also fixes the back navigation from the Alerts page to the Hosts view, this was the bug:

https://github.com/elastic/kibana/assets/12370520/a18b3ec0-b59b-441f-9d1c-65de2d3f19d0


### 🧪 How to test
**Please check the possible navigation scenarios related to the alerts page and its filters carefully**.
- Go to the alerts page by clicking on the right-hand side menu and navigate back
- Go to the alerts page, change a filter/time range, and navigate back
- On the alerts page, use brush functionality on the alert summary widget and navigate back
- Also, please test the scenarios mentioned in this [PR](https://github.com/elastic/kibana/pull/162312)

#### Known issue
If the time range is changed, the back navigation does not work properly. We have a similar case in APM, so I created a separate ticket for it: https://github.com/elastic/kibana/issues/186303
